src/Aspire.Hosting.Azure/AzureOpenAIResource.cs

```diff
@@ -0,0 +1,40 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.ObjectModel;
+
+namespace Aspire.Hosting.ApplicationModel;
+
+/// <summary>
+/// Represents an Azure OpenAI resource.
+/// </summary>
+/// <param name="name">The name of the resource.</param>
+public class AzureOpenAIResource(string name) : Resource(name), IAzureResource, IResourceWithConnectionString
+{
+    private readonly Collection<AzureOpenDeploymentResource> _deployments = [];
+
+    /// <summary>
+    /// Gets or sets the connection string for the Azure OpenAI resource.
+    /// </summary>
+    public string? ConnectionString { get; set; }
```

Comment:
Who sets this connection string? I don't see it being set. Do we expect the user to set this explicitly?

I guess there is this code that will find the connection info from the AppHost config if it isn't set:

https://github.com/dotnet/aspire/blob/3158bb1e893f9180babe2302468a29a83fca6582/src/Aspire.Hosting/Extensions/ResourceBuilderExtensions.cs#L165-L166

Reply:
For other Azure resources this is defined by the provisioner (and only it), but we don't have a provisioner here (yet). However implementing `IResourceWithConnectionString` is required to use `WithReference`. I could return `null`, or keep `ConnectionString` for when we have a provisioner.

Reply:
I think what you have now is fine. I was confused initially, but now I'm understanding it more.